cdap-app-templates/cdap-etl/cdap-etl-data-pipeline/src/test/java/co/cask/cdap/etl/datapipeline/DataPipelineTest.java

```diff
@@ -169,6 +173,40 @@ public void testSimpleMultiSource() throws Exception {
   }
 
   @Test
+  public void testProgramDescriptions() throws Exception {
```

Comment:
deploying an app seems like overkill for this, how about we the method package private and just call it directly in a test for BatchPhaseSpec